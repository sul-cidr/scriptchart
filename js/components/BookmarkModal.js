import React from "react";
import Modal from "react-modal";

import { CopyToClipboard } from "react-copy-to-clipboard";

Modal.setAppElement("#content");

class BookmarkModal extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      copied: false
    };
  }

  render() {
    return (
      <Modal
        isOpen={this.props.isOpen}
        onRequestClose={this.props.closeModal}
        shouldCloseOnOverlayClick={true}
        className="Modal"
        overlayClassName="Overlay"
        contentLabel="Bookmark URL"
      >
        <div className={"modal_header"}>
          <b>Bookmark</b>
          <button className={"button is-small"} onClick={this.props.closeModal}>
            Close
          </button>
        </div>
        <div className={"field box"}>
          <div className={"control"}>
            <div className="bookmarktext">
              {this.props.bookmarkURL}
            </div>
            <CopyToClipboard
              text={this.props.bookmarkURL}
              onCopy={() => this.setState({ copied: true })}
            ><div className={"box has-text-centered"}>
                <button className={"button is-info"}>Copy to clipboard</button>
              </div>
            </CopyToClipboard>
          </div>

          {this.state.copied ? (
            <p className={"has-text-success is-large"}>Copied.</p>
          ) : null}
        </div>
      </Modal>
    );
  }
}

export default BookmarkModal;
