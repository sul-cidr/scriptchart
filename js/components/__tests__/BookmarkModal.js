/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from "react";
import renderer from "react-test-renderer";
import BookmarkModal from "../BookmarkModal";

describe("BookmarkModal test", () => {
  it("BookmarkModal should match snapshot", () => {
    const component = renderer.create(<BookmarkModal />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});