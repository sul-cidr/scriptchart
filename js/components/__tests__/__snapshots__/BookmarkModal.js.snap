// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkModal test BookmarkModal should match snapshot 1`] = `null`;
